src/index.js

```diff
@@ -1,10 +1,18 @@
-/**
- * @param {number} lineNumber - zero based.
- * @return {number[]}
- */
-
 const pascalTriangle = (lineNumber) => {
-  
+  function pascalRecursive(n, array) {
+    if (n < 1) return array;
+    const prevTier = array[array.length - 1];
+    const curTier = [1];
+    for (let i = 1; i < prevTier.length; i++) {
+      curTier[i] = prevTier[i] + prevTier[i - 1];
+    }
+    curTier.push(1);
+    array.push(curTier);
+    return pascalRecursive(n - 1, array);
+  }
```

Comment:
Esto lo puedes pasar a un ciclo for, no veo que se requiera la recursividad más allá para iterar.
Un ejemplo de recursividad sería:
```js
const pascalRecursive = (n) => {
// base 
   if( n == 0) {
       return [[1]];
   }
 // suponemos que tenemos el anterior
  const triangle = pascalRecursive(n-1);
 // construimos la nueva respuesta 
  const currentTier = [1];
  for (let i = 1; i < n; i+= 1 ) {
     currentTier[i] =  triangle[n-1][i] + triangle[n-1][i-1];
  }
 currentTier.push(1);
  return [...triangle, currentTier];
}
```